lib/validate.js

```diff
@@ -2,19 +2,39 @@ var merge = require('mout/object/merge')
 var semver = require('semver')
 var subjectExceptions = /^(fixup|squash)!|^\[[^\]]+\]:/
 var defaults = {
-    component: true,
-    components: [],
-    limits: {
-      subject: 72,
-      other: 80
-    }
+  component: true,
+  components: [],
+  markerPattern: /^(clos|fix|resolv)(e[sd]|ing)/i,
+  actionPattern: /^([Cc]los|[Ff]ix|[Rr]esolv)(e[sd]|ing)\s+[^\s\d]+(\s|$)/,
+  ticketPattern: /^(Closes|Fixes) (.*#|gh-|[A-Z]{2,}-)[0-9]+/,
+  ticketPrefixes: /(.*#|gh-|[A-Z]{2,}-)/,
+  limits: {
+    subject: 72,
+    other: 80
+  }
 }
 
-module.exports = function (message, options) {
+function compose (defaults, options) {
   options = merge(defaults, options)
 
+  var prefixesPattern = new RegExp(options.ticketPrefixes)
+
+  options.ticketPattern = new RegExp(
```

Comment:
This makes sense when the original `options` argument (before merge) contains `ticketPrefixes`, but doesn't contain `ticketPattern`.